tests/tests/test-gas/test-gas-foreign-asset.ts

```diff
@@ -0,0 +1,163 @@
+    //   await createTransaction(context, {
+    //     ...CHARLETH_TRANSACTION_TEMPLATE,
+    //     to: ADDRESS_ERC20,
+    //     data: ERC20_INTERFACE.encodeFunctionData("approve", [alith.address, 100000000]),
+    //   })
+    // );
+
+    // const receipt = await context.web3.eth.getTransactionReceipt(result.hash);
+    // console.log("Zero approve gas used", receipt.gasUsed);
+    // expect(receipt.status).to.equal(true);
+
+    // ADD BALANCE TO ALITH
+    const assetId = context.polkadotApi.createType("u128", ASSET_ID);
+    // Get keys to modify balance
+    let module = xxhashAsU8a(new TextEncoder().encode("Assets"), 128);
+    let account_key = xxhashAsU8a(new TextEncoder().encode("Account"), 128);
+    let blake2concatAssetId = new Uint8Array([
+      ...blake2AsU8a(assetId.toU8a(), 128),
+      ...assetId.toU8a(),
+    ]);
+
+    let blake2concatAccount = new Uint8Array([
+      ...blake2AsU8a(hexToU8a(ALITH_ADDRESS), 128),
+      ...hexToU8a(ALITH_ADDRESS),
+    ]);
+    let overallAccountKey = new Uint8Array([
+      ...module,
+      ...account_key,
+      ...blake2concatAssetId,
+      ...blake2concatAccount,
+    ]);
+
+    // Get keys to modify total supply & dummyCode (TODO: remove once dummy code inserted by node)
+    let assetKey = xxhashAsU8a(new TextEncoder().encode("Asset"), 128);
+    let overallAssetKey = new Uint8Array([...module, ...assetKey, ...blake2concatAssetId]);
+    let evmCodeAssetKey = context.polkadotApi.query.evm.accountCodes.key(
+      "0xFfFFfFff" + assetId.toHex().slice(2)
+    );
+
+    const balance = new BN("100000000000000");
+    const assetBalance = context.polkadotApi.createType("PalletAssetsAssetAccount", {
+      balance: balance,
+    });
+    const assetDetails = context.polkadotApi.createType("PalletAssetsAssetDetails", {
+      supply: balance,
+    });
+
+    await context.createBlock(
+      context.polkadotApi.tx.sudo
+        .sudo(
+          context.polkadotApi.tx.system.setStorage([
+            [u8aToHex(overallAccountKey), u8aToHex(assetBalance.toU8a())],
+            [u8aToHex(overallAssetKey), u8aToHex(assetDetails.toU8a())],
+            [
+              evmCodeAssetKey,
+              `0x${((DUMMY_REVERT_BYTECODE.length - 2) * 2)
+                .toString(16)
+                .padStart(2)}${DUMMY_REVERT_BYTECODE.slice(2)}`,
+            ],
+          ])
+        )
+        .signAsync(alith)
+    );
+  });
+
+  it("should succeed when modifying approve", async function () {
+    context.ethTransactionType = "EIP1559";
+
+    const { result } = await context.createBlock(
+      await createTransaction(context, {
+        ...ALITH_TRANSACTION_TEMPLATE,
+        to: ADDRESS_ERC20,
+        data: ERC20_INTERFACE.encodeFunctionData("approve", [baltathar.address, 100000000]),
+      })
+    );
+
+    const receipt = await context.web3.eth.getTransactionReceipt(result.hash);
+    console.log("First approve gas used", receipt.gasUsed);
```

Comment:
Shouldn't we avoid console.logging inside TS tests? I think we had console.log() related issues in our CI in the past (correct me if I'm wrong).
```suggestion
```